src/app/util/attribute-storage.cpp

```diff
@@ -173,6 +173,18 @@ uint16_t emberAfIndexFromEndpointIncludingDisabledEndpoints(EndpointId endpoint)
     return findIndexFromEndpoint(endpoint, false /* ignoreDisabledEndpoints */);
 }
 
+constexpr const DeviceTypeId kRootnodeId   = 0x0016;
+constexpr const DeviceTypeId kAggregatorId = 0x000E;
+
+bool IsFullFamilyComposition(const EmberAfDeviceType & deviceType)
+{
+    if ((deviceType.deviceTypeId == kRootnodeId) || (deviceType.deviceTypeId == kAggregatorId))
+    {
+        return true;
+    }
+    return false;
```

Comment:
```suggestion
    return (deviceType.deviceTypeId == kRootnodeId) || (deviceType.deviceTypeId == kAggregatorId);
```

is there a spec reference that we can add here to explain?

Reply:
How is this composition used in other places today? Do we have no certification tests or do apps set composition manually?

If we auto-know what composition things should be, why do we even have the bitmap? Why don't we just return the composition type by looking at device types and never store it as a bitmap?

Reply:
> do apps set composition manually?

Even if they did so, it would produce no effect. `GetCompositionForEndpointIndex` would always return flat (cause of the bitmap values).
The device composition tests will always fail for a device having a chain with node length > 2. Example, Oven is parent of Cooktop which is parent of Cooksurface.


> If we auto-know what composition things should be, why do we even have the bitmap?

This is the alternative, let SDK decide what the composition should be instead of apps setting them.